lib/engineering/analysis/footing/centric_isolated.rb

```diff
@@ -0,0 +1,50 @@
+require 'errors/engineering/analysis/section_direction_error'
+
+module Engineering
+  module Analysis
+    module Footing
+      class CentricIsolated
+        ORTHOGONALITIES = %i[length_1 length_2].freeze
+
+        def initialize(footing:, load_from_column:, section_direction:)
+          if ORTHOGONALITIES.none?(section_direction)
+            raise Engineering::Analysis::SectionDirectionError.new(section_direction, ORTHOGONALITIES)
+          end
+
+          @footing = footing
+          @load_from_column = load_from_column
+          @section_direction = section_direction
+        end
+
+        def solicitation_load
+          solicitation * @footing.public_send(orthogonal_direction)
+        end
+
+        def max_shear_solicitation
+          solicitation_load * @footing.public_send(@section_direction)
```

Comment:
This `@footing.public_send(@section_direction)` call seems to be repeated many times in these methods. Maybe can you DRY it out by creating another `section_length` method that does this call? And same for the orthogonal length

```ruby
def section_length
  @footing.public_send(@section_direction)
end

def orthogonal_length
  @footing.public_send(orthogonal_direction)
end
```

Reply:
I like it, done!